app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -844,7 +844,12 @@ class FormController(
         } else {
           routes.FormController.form(formTemplateId, None, sn, sectionTitle4Ga, SuppressErrors.Yes, FastForward.Yes)
         }
-      case None => routes.SummaryController.summaryById(formTemplateId, maybeAccessCode, None)
+      case None =>
+        formTemplate.formKind match {
+          case FormKind.TaskList(_) =>
+            uk.gov.hmrc.gform.tasklist.routes.TaskListController.landingPage(formTemplateId, maybeAccessCode)
+          case _ => routes.SummaryController.summaryById(formTemplateId, maybeAccessCode, None)
+        }
```

Comment:
There is `fold` supposed to be used instead of patter matching on `formKind`
```suggestion
        formTemplate.formKind.fold { _ =>
          routes.SummaryController.summaryById(formTemplateId, maybeAccessCode, None)
        } { _ =>
          uk.gov.hmrc.gform.tasklist.routes.TaskListController.landingPage(formTemplateId, maybeAccessCode)
        }
```